sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.annotation.ServiceClient;
+
+import java.util.List;
+import java.util.Map;
+
+@ServiceClient(
+    builder = TableClientBuilder.class)
+public class TableClient {
+    String tableName;
+
+    TableClient(String tableName) {
+        this.tableName = tableName;
+    }
+
+    /**
+     * returns the table associated with this TableClient
+     *
+     * @param tableName name of the table
+     * @return the AzureTable
+     */
+    public AzureTable getTable(String tableName) {
```

Comment:
I don't think this has value, since the only things we can get from an `AzureTable` are its name, which we already have (`tableName`) and a `TableClient` for the table, which this is. I would just remove this method.